Lesson 16/task-1.py

```diff
@@ -0,0 +1,49 @@
+from abc import abstractmethod
+
+
+class Person:
+    def __init__(self, first_name, last_name):
+        self.first_name = first_name
+        self.last_name = last_name
+
+    def full_name(self):
+        return f'{self.first_name} {self.last_name}'
+
+    @abstractmethod
+    def eat(self):
+        pass
+
+
+class Teacher(Person):
+    def __init__(self, first_name, last_name, salary, subject):
```

Comment:
It makes sense to pass arguments like `*args`. This way you won't need to rewrite args in child classes every time you change them in a parent class.